lib/logstash/codecs/multiline.rb

```diff
@@ -177,14 +183,19 @@ def flush(&block)
 
   def do_next(text, matched, &block)
     buffer(text)
-    flush(&block) if !matched
+    flush(&block) if !matched || maximum_buffered_lines?
   end
 
   def do_previous(text, matched, &block)
-    flush(&block) if !matched
+    flush(&block) if !matched || maximum_buffered_lines?
     buffer(text)
   end
 
+
+  def maximum_buffered_lines?
+    @buffer.size > @max_lines
```

Comment:
+1 agree, What would be the best way? maybe adding a tag to the generated event?